server/src/controllers/Events/resolver.ts

```diff
@@ -230,44 +229,36 @@ const getNameForNewAttendance = (event: EventAttendanceName) => {
 
 @Resolver()
 export class EventResolver {
-  @Query(() => PaginatedEventsWithTotal)
+  @Query(() => [PaginatedEventsWithChapters])
   async paginatedEventsWithTotal(
     @Arg('limit', () => Int, { nullable: true }) limit?: number,
     @Arg('offset', () => Int, { nullable: true }) offset?: number,
-  ): Promise<PaginatedEventsWithTotal> {
-    const total = await prisma.events.count();
-    const events = await prisma.events.findMany({
-      include: {
-        chapter: true,
-      },
-      orderBy: [{ start_at: 'asc' }, { name: 'asc' }],
-      take: limit ?? 10,
-      skip: offset,
+    @Arg('showOnlyUpcoming', () => Boolean, { nullable: true })
+    showOnlyUpcoming = true,
+  ): Promise<PaginatedEventsWithChapters[]> {
+    const total = await prisma.events.count({
+      ...(showOnlyUpcoming && {
+        where: {
+          canceled: false,
+          ends_at: { gt: new Date() },
+        },
+      }),
     });
-    return { total, events };
-  }
-
-  @Query(() => [EventWithChapter])
-  async paginatedEvents(
-    @Arg('limit', () => Int, { nullable: true }) limit?: number,
-    @Arg('offset', () => Int, { nullable: true }) offset?: number,
-  ): Promise<EventWithChapter[]> {
-    return await prisma.events.findMany({
-      where: {
-        AND: [
-          {
-            canceled: false,
-            ends_at: { gt: new Date() },
-          },
-        ],
-      },
+    const events = await prisma.events.findMany({
+      ...(showOnlyUpcoming && {
+        where: {
+          canceled: false,
+          ends_at: { gt: new Date() },
+        },
+      }),
       include: {
         chapter: true,
       },
       orderBy: [{ start_at: 'asc' }, { name: 'asc' }],
       take: limit ?? 10,
       skip: offset,
     });
+    return [{ total, events }];
```

Comment:
```suggestion
    return { total, events };
```

Reply:
@Sboonny I did try this locally and I'm completely at a loss as to why it fails.

The GQL query works just fine in the studio, but for some reason the client gets `data.paginatedEventsWithTotal === []`. This makes no sense to me.

Unless either of you have any clue why this is failing, I think we can just add an explanatory comment and be done with it.

Reply:
That's due to query policy merging the results
https://github.com/Sboonny/chapter/blob/80f58331fee00649361e936bba9b01ad680cfc78/client/src/pages/_app.tsx#L61

For the changed resolver something like this works, and it actually merges them flattened, which is pretty cool.
```typescript
          paginatedEventsWithTotal: {
            keyArgs: ['showOnlyUpcoming'],
            merge(existing = { total: undefined, events: [] }, incoming) {
              return {
                total: existing.total ?? incoming.total,
                events: [...existing.events, ...incoming.events],
              };
            },
          },
```

Reply:
Oh, thanks, I completely missed that!